lib/envelope.c

```diff
@@ -106,7 +106,7 @@ static int _ve_amp(envelope_lookup *ve,
   /* stretch is used to gradually lengthen the number of windows
      considered prevoius-to-potential-trigger */
   int stretch=max(VE_MINSTRETCH,ve->stretch/2);
-  float penalty=gi->stretch_penalty-(ve->stretch/2-VE_MINSTRETCH);
+  float penalty=gi->stretch_penalty-((float)(ve->stretch)/2-VE_MINSTRETCH);
```

Comment:
The stretch itself is calculated with integer division, so maybe that's a more accurate heuristic for the penalty?

Reply:
you know better